### PR TITLE
Added support for DataTypeAttribute

### PR DESCRIPTION
**Pull request by nickcox**

### CODE REVIEW

File: src/NJsonSchema/Generation/JsonSchemaGenerator.cs [hunk 1]
```diff
@@ -531,6 +539,27 @@ public void ApplyPropertyAnnotations(JsonSchema4 jsonProperty, Type parentType,
                     jsonProperty.MaxLength = stringLengthAttribute.MaximumLength;
                 }
             }
+
+            dynamic dataTypeAttribute = TryGetAttribute(attributes,
+                "System.ComponentModel.DataAnnotations.DataTypeAttribute");
+            if (dataTypeAttribute != null)
+            {
+                var knownFormats = new Dictionary<string, string>
```
Comment (rsuter) [inline src/NJsonSchema/Generation/JsonSchemaGenerator.cs]: You should make this static in the class so that it is not created multiple times...

**Comment by rsuter** [comment]
Looks good so far, check my comment on line 547

**Comment by rsuter** [comment]
You should also use the consts from here: 

https://github.com/NJsonSchema/NJsonSchema/blob/master/src/NJsonSchema/JsonFormatStrings.cs

Are your "format" strings official (ie. "data-url" and "phone")?

**Comment by rsuter** [comment]
Also, we should either implement validation for these new format types or add an issue for this...

**Comment by nickcox** [comment]
AFAICT the [official format strings](http://json-schema.org/latest/json-schema-validation.html#anchor107) are:

* date-time
* email
* hostname
* ipv4
* ipv6
* uri

However, custom formats are [allowed](http://json-schema.org/latest/json-schema-validation.html#anchor106) and these all seem to be supported by at least [some libraries](https://github.com/mozilla-services/react-jsonschema-form). I guess if we need to implement corresponding validations then that's an argument for not including the unofficial ones.

**Comment by rsuter** [comment]
I think we should only validate the official ones and generate the formats defined by Json.NET

**Comment by nickcox** [comment]
Makes sense.

**Comment by rsuter** [comment]
Added new issue: https://github.com/NJsonSchema/NJsonSchema/issues/184

**Pull request merged**